app/models/partner.rb

```diff
@@ -235,6 +235,10 @@ def invite_new_partner
   end
 
   def should_invite_because_email_changed?
-    email_changed? and (invited? or awaiting_review? or recertification_required? or approved?)
+    email_changed? and (invited? or awaiting_review? or recertification_required? or approved?) and !partner_user_with_same_email_exist?
```

Comment:
I'm not sure why this was previously implemented with `and`/`or` instead of `&&`/`||`, but it might be worth changing it to `&&`/`||` since we're in here anyway